lib/console.js

```diff
@@ -35,20 +37,40 @@ function Console(stdout, stderr) {
   }
 }
 
+function write(ignoreErrors, stream, string) {
+  if (!ignoreErrors) return stream.write(string);
+
+  // There may be an error occurring synchronously (e.g. for files or TTYs)
+  // or asynchronously (e.g. pipes), so handle both situations.
+  try {
+    stream.write(string, (err) => {
+      if (err) {
+        // If there was an error, it will be emitted on `stream` as
+        // an `error` event. Adding a `once` listener will keep that error
+        // from becoming an uncaught exception, but since the handler is
+        // removed after the event, non-console.* writes won’t be affected.
+        stream.once('error', noop);
```

Comment:
https://github.com/nodejs/node/pull/9471#pullrequestreview-11489528 … buffering or anything like that shouldn’t be the problem. But maybe this doesn’t work when the callback is called synchronously? I’d have to try it out but maybe @mcollina knows an answer off the top of his head?